sunkit_image/coalignment/match_template.py

```diff
@@ -0,0 +1,130 @@
+from typing import NamedTuple
+
+import astropy.units as u
+import numpy as np
+from skimage.feature import match_template
+
+from sunkit_image.utils.decorators import register_coalignment_method
+
+__all__ = ["match_template_coalign"]
+
+
+class affine_params(NamedTuple):
```

Comment:
This should be moved to the interface file or elsewhere since its going to be used elsewhere in the future.

Reply:
Also I don't think we need to subclass namedtuple or define the types for it.

Reply:
Agreed. Also I think Will was suggesting instead of this we could return a affine matrix with all the parameters. I'll make the changes.

Reply:
I think I sort of disagree with Nabil on adding the types here (which I didn't expect :laughing:) I didn't actually know you could do this with a `NamedTuple`. I would like to see this as is or as a dataclass as I think we will want to document this thoroughly (including types, and docstrings for all the attributes).

So using this form we can add docstrings:

````python
class affine_params(NamedTuple):
    scale: tuple[tuple[float, float], tuple[float, float]]
    """
    This is the docstring for scale.
    """
[...]
```

I do not think we should return a single affine transform matrix, as that moves us too far away from FITS WCS to make it easier to process afterwards. I would like to see this return scale, a rotation matrix (2x2) and a translation.

Reply:
> I think I sort of disagree with Nabil on adding the types here (which I didn't expect 😆) I didn't actually know you could do this with a `NamedTuple`. I would like to see this as is, or as a dataclass, as I think we will want to document this thoroughly (including types, and docstrings for all the attributes).
> 
> So using this form we can add docstrings:
> 
> ```python
> class affine_params(NamedTuple):
>     scale: tuple[tuple[float, float], tuple[float, float]]
>     """
>     This is the docstring for scale.
>     """
> [...]
> ```

Sure :)

> I do not think we should return a single affine transform matrix, as that moves us too far away from FITS WCS to make it easier to process afterwards. I would like to see this return scale, a rotation matrix (2x2) and a translation.

That makes sense. I'll add the documentation.